### ISSUE TITLE
Improve outline view

### ISSUE THREAD

**Issue by mads-hartmann** [mads-hartmann/bash-language-server#25]
Currently the outline view simply shows a flat list of all symbols in the file. I believe that you can get a hierarchical view if you return a bit more information (I think it's the name of the parent)

**Comment by rcjsuen** [comment]
This isn't really supported in the [current specification](https://github.com/Microsoft/language-server-protocol/blob/f09e61c23d57213bbeae8ff329ea4f856611b15d/specification.md). Also see Microsoft/language-server-protocol#327 and its many referenced issues and so on.

```TypeScript
/**
 * The name of the symbol containing this symbol. This information is for
 * user interface purposes (e.g. to render a qualifier in the user interface
 * if necessary). It can't be used to re-infer a hierarchy for the document
 * symbols.
 */
containerName?: string;
```

**Comment by laughedelic** [comment]
Notice that the parent name is not enough. See discussion in https://github.com/facebook-atom/atom-ide-ui/issues/121 and https://github.com/facebook-atom/atom-ide-ui/issues/121#issuecomment-344145808 in particular for the summary.

**TL;DR**: LSP doesn't allow you to rely on `containerName` for the hierarchy. But you can use nested location ranges to nest the tree nodes.

**Comment by mads-hartmann** [comment]
@laughedelic Thanks a lot, that was super helpful 👍  I've opened a PR #31 now which seems to do the trick in both Atom and Code. I will probably merge if early next week and cut a release with some other new features as well :)

**Comment by mads-hartmann** [comment]
We won't improve the outline view further until it's a proper part of the LSP definition.
